cpmpy/expressions/core.py

```diff
@@ -272,21 +277,25 @@ def __rmul__(self, other):
     # other mathematical ones
     def __truediv__(self, other):
         warnings.warn("We only support floordivision, use // in stead of /", SyntaxWarning)
-        if is_num(other) and other == 1:
-            return self
-        return Operator("div", [self, other])
+        return self.__floordiv__(other)
+
     def __rtruediv__(self, other):
         warnings.warn("We only support floordivision, use // in stead of /", SyntaxWarning)
-        return Operator("div", [other, self])
+        return self.__rfloordiv__(other)
+
     def __floordiv__(self, other):
         if is_num(other) and other == 1:
             return self
         return Operator("div", [self, other])
+
     def __rfloordiv__(self, other):
         return Operator("div", [other, self])
 
     def __mod__(self, other):
+        if is_num(other) and other == 1:
+            return self
```

Comment:
This should be `return 0` - anything mod 1 is 0?

Reply:
Or just drop the optimization. This kind of optimization may be better in an actual transformation instead of sprinkled throughout the code (though it is tempting! :D )

Reply:
Maybe this is why the test fails?

Reply:
ho my bad yes... mod 1 = 0... This is what copy-paste leads to.. I'm updating the PR

Reply:
I also agree that this kind of optimization should be done elsewhere (linked to issue #334